snuba/admin/static/runtime_config/auditlog.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { ReactNode, useState } from "react";
+
+import Client from "../api_client";
+import { containerStyle, paragraphStyle } from "./styles";
+import { Table } from "../table";
+import { ConfigValue } from "./types";
+
+function AuditLog(props: { api: Client }) {
+  const { api } = props;
+  const [data, setData] = useState<any[] | null>(null);
+
+  function fetchData() {
+    api.getAuditlog().then((res) => {
+      setData(res);
+    });
+  }
+
+  if (data === null) {
+    fetchData();
+  }
+
+  if (!data) {
+    return null;
+  }
+
+  const rowData = data.map(({ key, timestamp, user, before, after }) => {
+    const formattedDate = new Date(timestamp * 1000).toLocaleDateString(
+      "en-US",
+      {
+        day: "numeric",
+        month: "short",
+        year: "numeric",
+        hour: "numeric",
+        minute: "numeric",
+        second: "numeric",
+        timeZoneName: "short",
+      }
+    );
+
+    return [
+      formattedDate,
+      user,
+      <code>{key}</code>,
+      getActionDetail(before, after),
+    ];
+  });
+
+  const usersTimeZone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+
+  return (
+    <div style={containerStyle}>
+      <p style={paragraphStyle}>
+        Only changes to runtime configuration are currently being captured in
+        the audit log
+      </p>
+      <Table
+        headerData={[`Timestamp (${usersTimeZone})`, "User", "Key", "Action"]}
+        rowData={rowData}
+        columnWidths={[1, 1, 1, 1]}
+      ></Table>
+    </div>
+  );
+}
+
+function getActionDetail(
+  before: ConfigValue | null,
+  after: ConfigValue | null
+): ReactNode {
+  const formattedBefore = typeof before === "string" ? `"${before}"` : before;
+  const formatttedAfter = typeof after === "string" ? `"${after}"` : after;
+
+  if (before === null && after !== null) {
+    if (after === null) {
+      // Value was null before and after the change, something went wrong
+      throw new Error("Unkknown action");
```

Comment:
should this really throw or just make the error text something like 

❗ Value was null before and after the change, something went wrong ❗

Reply:
because if we throw the whole page won't render right?

Reply:
Yeah, I want to add a component to render the error text properly on all of our pages if the API can't load or comes back with totally invalid data. I didn't get to it yet but still thought it's better to fail hard than quietly for now.

Reply:
oh also, this is dead code,

```javascript
  if (before === null && after !== null) {
    if (after === null) {
      // dead code
    }
  }

Reply:
Oops, thanks. Fixed.